examples/config.example.json

```diff
@@ -1,61 +1,83 @@
-            "skip_check_package_names": true
-        },
-        "release": {
-            "has_nightly_track": false,
-            "service_account": "service-account-which-have-access-to-release@iam.gserviceaccount.com",
-            "certificate": "/absolute/path/to/certificate-for-release-service-account.p12",
-            "update_google_play_strings":  true,
-            "digest_algorithm": "SHA1",
-            "skip_check_package_names": true
-        },
-        "focus": {
-            "has_nightly_track": false,
-            "service_account": "service-account-which-have-access-to-focus@iam.gserviceaccount.com",
-            "certificate": "/absolute/path/to/certificate-for-release-service-account.p12",
-            "update_google_play_strings":  true,
-            "digest_algorithm": "SHA-256",
-            "expected_package_names": ["org.mozilla.klar", "org.mozilla.focus"],
-            "skip_check_ordered_version_codes":  true,
-            "skip_checks_fennec": true
-        },
-        "fenix-beta": {
-            "require_track": "beta",
-            "service_account": "service-account-fenix-beta@iam.gserviceaccount.com",
-            "certficiate": "/absolute/path/to/certificate-for-fenix-beta-service-account.p12",
-            "update_google_play_strings": false,
-            "digest_algorithm": "SHA-256",
-            "expected_package_names": ["org.mozilla.fenix.beta"],
-            "skip_check_multiple_locales": true,
-            "skip_check_same_locales": true,
-            "skip_checks_fennec": true
+    "products": [{
+        "product_names": ["aurora", "beta", "release"],
+        "update_google_play_strings": true,
+        "digest_algorithm": "SHA1",
+        "use_scope_for_channel": true,
+        "map_channels_to_apps": true,
+        "apps": {
+            "aurora": {
+                "package_names": ["org.mozilla.fennec_aurora"],
+                "google_play_track": "beta",
+                "service_account": "firefox-aurora@iam.gserviceaccount.com",
+                "google_credentials_file": "/firefox-nightly.p12",
+                "certificate_alias": "nightly"
+            },
+            "beta": {
+                "package_names": ["org.mozilla.firefox_beta"],
+                "google_play_track": "production",
+                "service_account": "firefox-beta@iam.gserviceaccount.com",
+                "google_credentials_file": "/firefox.p12",
+                "certificate_alias": "release"
+            },
+            "release": {
+                "package_names": ["org.mozilla.firefox"],
+                "google_play_track": "production",
+                "service_account": "firefox-production@iam.gserviceaccount.com",
+                "google_credentials_file": "/firefox.p12",
+                "certificate_alias": "release"
+            }
         }
-    },
+    },{
+        "product_names": ["focus"],
+        "update_google_play_strings": false,
+        "digest_algorithm": "SHA-256",
```

Comment:
Follow up: Let's make sure `digest_algorithm` is within a known set of values and bail out early if it doesn't match any. I'm thinking of the case of `SHA256` (instead of `SHA-256`) => we would need to wait for a push-apk task to run, and then we would be super confused because `SHA256` isn't in the output of `jarsigner`.

Reply:
I agree that it's good to avoid situations where debugging would be challenging due to missing information.

However, I'm not sure that we should enforce a set of valid digest algorithms. If the behaviour of `pushapkscript` itself was limited to certain algorithms (e.g.: we could handle `SHA1` and `SHA256`, but literally didn't have an implementation for, say, `MD5`), then I would absolutely agree that we should validate against that. I think a good example of this is `map_channels_to_tracks` - if its `True`, but `single_app_config` isn't set, then `pushapkscript` _physically_ cannot complete: that's invalid config.

However, the purpose of `digest_algorithm` in `pushapkscript` is to ensure that the algorithm doesn't change between publications. We don't actually care what the value is, we just want it to be rigid. Theoretically, "Fenix 2" could use `SHA512`, and the implementation of `pushapkscript` isn't actually affected.

For this reason, I'm thinking that it doesn't make sense to hard-code a set of "valid" algorithms. However, I will create a ticket to ensure that, when the algorithm doesn't match, we make it more obvious why ([created ticket](https://github.com/mozilla-releng/pushapkscript/issues/90))

What do you think?

Reply:
That sounds totally valid to me. Let's make the error more obvious like you described in https://github.com/mozilla-releng/pushapkscript/issues/90. Thanks!